### PR TITLE
Support Multiple Projects

### PR DESCRIPTION
**Pull request by shaunanoordin**
## PR Overview

This PR is a major modification to the Community Catalog - the website can now catalogue subjects from **multiple projects,** instead of just one Zooniverse project. (The list of projects has to be preconfigured in advance.)

- ❗ New behaviour: **multi-project support**
  - Available projects are now configured via **projects.json**
  - New URL format: projects are now accessed by `https://local.zooniverse.org:8080/projects/PROJECT_OWNER/PROJECT_NAME` (i.e. the same project slug format as zooniverse.org)
  - e.g. the Scarlets & Blues project home page is `https://local.zooniverse.org:8080/projects/bogden/scarlets-and-blues?env=production`
  - ⚠️ We are missing a home page URL, i.e. `https://local.zooniverse.org:8080/` just errors out now.
- New components:
  - ProjectContainer: handles active project selection
- New dependencies: MobX, MobX React, MobX State Tree
- Code changes: internal stores
  - `project` (active project) and `user` resources now moved to the stores, as they may be required across multiple components.
  - all other variables remain in their local states, where possible. (e.g. App's `initialised`)

### CODE REVIEW

**Comment by shaunanoordin** [comment]
Still on the TODO:

- [ ] Links need to be fixed
- [ ] A placeholder home page needs to be added. Current "home page" should be renamed "project home page"

File: src/App/App.js [hunk 1]
```diff
@@ -1,23 +1,24 @@
-import React, { useState, useEffect } from 'react'
+import React, { useState, useEffect, useContext } from 'react'
```
Comment (eatyourgreens) [inline src/App/App.js:1]: The `React` global was [removed from React](https://legacy.reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html) in 2020, so I'd avoid using it in new code.

Comment (shaunanoordin) [reply src/App/App.js]: Thanks! 👍 Redundant `import React`s have been removed.

**Comment by shaunanoordin** [comment]
## PR Updates

- Links fixed / more components use Project Data:
  - Header only displays links and search bar if a project is available
  - "Project Home" and "Talk Page" links now link to the appropriate pages on zooniverse.org 
- Helper functions fixed / use project data:
  - fetchKeywords() and fetchTagSearchResults() now require Project ID.
- Minor: unnecessary `import React`s removed

### Status

Ready to go!

**Pull request merged**